### PR TITLE
Replace console.warn/error with user logging

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -211,12 +195,12 @@ export class Log {
...
    * Creates an error object.
-   * @param {string|!Error} errorOrMessage
+   * @param {...*} var_args
    * @return {!Error}
    */
-  createError(errorOrMessage) {
-    const error = typeof errorOrMessage == 'string' ?
-        new Error(errorOrMessage) : errorOrMessage;
+  createError(var_args) {
+    const error = createErrorVargs.apply(null,
```
Comment (avimehta) [inline src/log.js]: createErrorVArgs?

**Comment by avimehta** [comment]
some tests fail but amp-analytics changes lgtm.

File: src/log.js (hunk 1 above)
Comment (dvoytenko) [inline src/log.js]: oops, this is from a different pr. got here accidentally. removed. though probably should fix there instead...

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

**Pull request merged**